core/channel.go

```diff
@@ -27,10 +28,6 @@ func CreateChannel(src, dst *ProvableChain, ordered bool, to time.Duration) erro
 			return err
 		}
 
-		if !chanSteps.Ready() {
-			break
-		}
```

Comment:
Could you keep this block and replace `break` with `continue` + some info logs ("waiting for next channel step ..." or something)?
Please do the same for `connection.go`.

Reply:
Sorry, debug log is more suitable than info log.
The finalization of some blockchains takes several minutes, which outputs too much info logs.

Reply:
In addition, it is helpful to know the state of the finalized and latest connection/channel.

Reply:
Added debug log 
https://github.com/hyperledger-labs/yui-relayer/pull/101/commits/b592e03420d2bb38aae3b007966074efb169d2c5
like
```
+ /Users/dongri.jin/go/src/github.com/datachainlab/toki-relayer-longrun/tests/eth2eth/scripts/../../../relayer/build/relayer --home /Users/dongri.jin/go/src/github.com/datachainlab/toki-relayer-longrun/tests/eth2eth/scripts/../demo/.urelayer tx connection ibc01
2023/08/30 15:24:33 - [ibc0]@{0}conn(connection-0)-{STATE_UNINITIALIZED_UNSPECIFIED} : [ibc1]@{0}conn(connection-0)-{STATE_UNINITIALIZED_UNSPECIFIED}
2023/08/30 15:24:40 Source Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:24:40 waiting for next connection step ...
2023/08/30 15:24:40 Source Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:24:40 waiting for next connection step ...
2023/08/30 15:24:41 Source Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:24:41 waiting for next connection step ...
2023/08/30 15:24:42 Source Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:24:42 waiting for next connection step ...
2023/08/30 15:24:43 Source Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:24:43 waiting for next connection step ...
2023/08/30 15:24:44 - [ibc1]@{0}conn(connection-0)-{STATE_UNINITIALIZED_UNSPECIFIED} : [ibc0]@{198}conn(connection-0)-{STATE_INIT}
2023/08/30 15:24:49 Destination Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:24:49 waiting for next connection step ...
2023/08/30 15:24:49 Destination Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:24:49 waiting for next connection step ...
2023/08/30 15:24:50 Destination Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:24:50 waiting for next connection step ...
2023/08/30 15:24:51 Destination Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:24:51 waiting for next connection step ...
2023/08/30 15:24:52 Destination Connection: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:24:52 waiting for next connection step ...
2023/08/30 15:24:53 - [ibc0]@{207}conn(connection-0)-{STATE_INIT} : [ibc1]@{208}conn(connection-0)-{STATE_TRYOPEN}
2023/08/30 15:25:00 Source Connection: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:00 waiting for next connection step ...
2023/08/30 15:25:00 Source Connection: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:00 waiting for next connection step ...
2023/08/30 15:25:01 Source Connection: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:01 waiting for next connection step ...
2023/08/30 15:25:02 Source Connection: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:02 waiting for next connection step ...
2023/08/30 15:25:03 Source Connection: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:03 waiting for next connection step ...
2023/08/30 15:25:04 - [ibc1]@{219}conn(connection-0)-{STATE_TRYOPEN} : [ibc0]@{218}conn(connection-0)-{STATE_OPEN}
2023/08/30 15:25:09 ★ Connection created: [ibc0]client{mock-client-0}conn{connection-0} -> [ibc1]client{mock-client-0}conn{connection-0}
+ /Users/dongri.jin/go/src/github.com/datachainlab/toki-relayer-longrun/tests/eth2eth/scripts/../../../relayer/build/relayer --home /Users/dongri.jin/go/src/github.com/datachainlab/toki-relayer-longrun/tests/eth2eth/scripts/../demo/.urelayer tx channel ibc01
2023/08/30 15:25:09 - [ibc0]@{0}chan(channel-0)-{STATE_UNINITIALIZED_UNSPECIFIED} : [ibc1]@{0}chan(channel-0)-{STATE_UNINITIALIZED_UNSPECIFIED}
2023/08/30 15:25:12 Source Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:25:12 waiting for next channel step ...
2023/08/30 15:25:13 Source Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:25:13 waiting for next channel step ...
2023/08/30 15:25:14 Source Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:25:14 waiting for next channel step ...
2023/08/30 15:25:15 Source Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_INIT]
2023/08/30 15:25:15 waiting for next channel step ...
2023/08/30 15:25:16 - [ibc1]@{0}chan(channel-0)-{STATE_UNINITIALIZED_UNSPECIFIED} : [ibc0]@{230}chan(channel-0)-{STATE_INIT}
2023/08/30 15:25:22 Destination Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:25:22 waiting for next channel step ...
2023/08/30 15:25:23 Destination Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:25:23 waiting for next channel step ...
2023/08/30 15:25:24 Destination Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:25:24 waiting for next channel step ...
2023/08/30 15:25:25 Destination Channel: Finalized state [STATE_UNINITIALIZED_UNSPECIFIED] <> Latest state [STATE_TRYOPEN]
2023/08/30 15:25:25 waiting for next channel step ...
2023/08/30 15:25:26 - [ibc0]@{240}chan(channel-0)-{STATE_INIT} : [ibc1]@{241}chan(channel-0)-{STATE_TRYOPEN}
2023/08/30 15:25:32 Source Channel: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:32 waiting for next channel step ...
2023/08/30 15:25:33 Source Channel: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:33 waiting for next channel step ...
2023/08/30 15:25:34 Source Channel: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:34 waiting for next channel step ...
2023/08/30 15:25:35 Source Channel: Finalized state [STATE_INIT] <> Latest state [STATE_OPEN]
2023/08/30 15:25:35 waiting for next channel step ...
2023/08/30 15:25:36 - [ibc1]@{251}chan(channel-0)-{STATE_TRYOPEN} : [ibc0]@{250}chan(channel-0)-{STATE_OPEN}
2023/08/30 15:25:42 ★ Channel created: [ibc0]chan{channel-0}port{transfer} -> [ibc1]chan{channel-0}port{transfer}
+ set +x
```